src/lib/providers.ts

```diff
@@ -1,21 +1,35 @@
-import { commands, languages, window, ExtensionContext } from 'vscode';
+import {
+  commands,
+  languages,
+  window,
+  ExtensionContext,
+  workspace,
+} from 'vscode';
 import { FileCompletionProvider } from './providers/fileCompletion';
 import { COMMANDS, TREE_DATA } from './constants';
 
 import { AccountsProvider } from './providers/treedata/accounts';
 import { HelpAndFeedbackProvider } from './providers/treedata/helpAndFeedback';
+import { RemoteFileProvider } from './providers/remoteFileProvider';
+import { RemoteFsProvider } from './providers/remoteFsProvider';
 
 const initializeTreeDataProviders = (context: ExtensionContext) => {
   const accountProvider = new AccountsProvider();
   const fileCompletionProvider = new FileCompletionProvider();
   const helpAndFeedbackProvider = new HelpAndFeedbackProvider();
+  const remoteFsProvider = new RemoteFsProvider();
+  const remoteFileProvider = RemoteFileProvider;
+  const scheme = 'hubl';
 
   context.subscriptions.push(
     commands.registerCommand(COMMANDS.ACCOUNTS_REFRESH, () => {
       console.log(COMMANDS.ACCOUNTS_REFRESH);
       accountProvider.refresh();
     })
   );
+  context.subscriptions.push(
+    workspace.registerTextDocumentContentProvider(scheme, remoteFileProvider)
+  );
```

Comment:
A `TextDocumentContentProvider` essentially tells VSCode how it should open items with different URIs. In this case we register our custom `remoteFileProvider` to the scheme 'hubl' so that we can use it to open items that begin `hubl:`